users/views.py

```diff
@@ -124,6 +125,7 @@ def user_login(request):
     if request.user.is_authenticated():
         return redirect(next_page)
     if request.method == 'POST':
+        deactivate_non_constant_public_users()
```

Comment:
If the user is authenticated, it's hard to let the user logout.
It's easier that we check before we validate the form